pkgs/http_client_conformance_tests/lib/src/redirect_tests.dart

```diff
@@ -20,7 +21,8 @@ void testRedirect(Client client, {bool redirectAlwaysAllowed = false}) async {
     late final StreamQueue<Object?> httpServerQueue;
 
     setUpAll(() async {
-      httpServerChannel = await startServer('redirect_server.dart');
+      httpServerChannel =
+          await startServer('redirect_server.dart', server.hybridMain);
```

Comment:
It feels unfortunate how references to target file (redirect_server.dart) are duplicated as a runtime parameter to a startServer function as well as static import as a target for `server.hybridMain` target function.
Why generic `spawnHybridUri` approach doesn't work for flutter plugin?

Reply:
The problem with flutter plugins is that the integration tests are run either on a device or on the emulator. But spawnHybridUri will start the server on the computer running the test and the two might not share the same `localhost`. So my change runs the server on the device itself, except in the web case where running the server in the browser is not possible.

If we don't like this approach, it might not be possible to have special logic for the android emulator where we hardcode the address of the host device (it's always `10.0.2.2` [1]). The iOS emulator uses the same network as the host. And we could not support running the tests on read devices.

[1] https://developer.android.com/studio/run/emulator-networking